enterprise/backend/src/metabase_enterprise/data_editing/data_editing.clj

```diff
@@ -123,14 +124,17 @@
     ;; TODO fix tests that execute actions without a user scope
     (when user-id
       (when-not (some (comp #{:data-editing/undo :data-editing/redo} first) invocation-stack)
-        ((requiring-resolve 'metabase-enterprise.data-editing.undo/track-change!)
+        (undo/track-change!
          user-id
          scope
          (u/for-map [[table-id diffs] (group-by :table-id diffs)]
-           [table-id (u/for-map [{:keys [before after] :as diff} diffs
+           [table-id (u/for-map [{:keys [before after undoable] :as diff} diffs
                                  :when (or before after)
                                  :let [{:keys [pk before after]} (diff->pk-diff diff)]]
-                       [pk [before after]])]))))
+                       [pk {:raw_before before
+                            :raw_after  after
+                            ;; default is true
+                            :undoable   (if (nil? undoable) true undoable)}])]))))
```

Comment:
I'd prefer to keep the "business logic" that we can't undo this stuff within the `undo` namespace. So basically we should pass in some kind of "fact" to the track method, rather than a flag.

The fact I was thinking of passing is the fact that the `effects` are incomplete - that additional rows were modified, but we don't know which ones, and (keeping this very general) how they were changed.

So rather than this being of property associated even with a particular `pk`, it is an additional sentinel `diff`.

Does that make sense? What do you think?

Reply:
`undoable` is a bit of a weird column, as ultimately it's a property of the batch, not of the row.

Reply:
A less drastic / more information preserving change than having the sentinel could be to have the "factual" key `deleted-chidren` on this `diff`.